compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -899,7 +904,7 @@ impl<'tcx> ty::FallibleTypeFolder<TyCtxt<'tcx>> for RemapHiddenTyRegions<'tcx> {
                 );
             }
         } else {
-            let guar = match region.opt_param_def_id(self.tcx, self.tcx.parent(self.def_id)) {
+            let guar = match region.opt_param_def_id(self.tcx, self.impl_m_def_id) {
```

Comment:
This used the def id of the trait, rather than the def id of the impl method. The def id of the impl method is needed here because we're remapping args *from* the impl method's signature.